libraries/AP_Scripting/examples/aileron_yaw_mix.lua

```diff
@@ -0,0 +1,197 @@
+
+   local _iMax = iMax
+   local _last_t = nil
+   local _log_data = {}
+   local _I = 0.0
+   local _counter = 0
+
+   local _pid_out = {}
+   _pid_out.FF = 0.0
+   _pid_out.P = 0.0
+   _pid_out.I = 0.0
+   _pid_out.target = 0.0
+   _pid_out.actual = 0.0
+
+   -- update the controller.
+   function self.update(target, current)
+      local now = millis():tofloat() * 0.001
+      if not _last_t then
+         _last_t = now
+      end
+
+      local dt = now - _last_t
+      _last_t = now
+      local err = target - current
+      _counter = _counter + 1
+
+      local FF = _kFF:get() * target
+      local P = _kP:get() * err
+      _I = _I + _kI:get() * err * dt
+      imax = _iMax:get()
+      _I = constrain(_I, -imax, imax)
+      local I = _I
+      local ret = FF + P + I
+
+      _log_data = { target, current, FF, P, I, ret }
+
+      _pid_out.FF = FF
+      _pid_out.P = P
+      _pid_out.I = I
+      _pid_out.target = target
+      _pid_out.actual = current
+
+      return ret
+   end
+
+   -- log the controller internals
+   function self.log(name)
+      logger.write(name,'Targ,Curr,FF,P,I,Total','ffffff',table.unpack(_log_data))
+   end
+
+   function self.send_telemetry()
+      gcs:send_named_float(string.format('YAW_FF'), _pid_out.FF)
+      gcs:send_named_float(string.format('YAW_P'), _pid_out.P)
+      gcs:send_named_float(string.format('YAW_I'), _pid_out.I)
+      gcs:send_named_float(string.format('YAW_TG'), math.deg(_pid_out.target))
+      gcs:send_named_float(string.format('YAW_AC'), math.deg(_pid_out.actual))
+   end
+
+   -- return the instance
+   return self
+end
+
+local yaw_pid = PIFF(AMIX_YAW_FF, AMIX_YAW_P, AMIX_YAW_I, AMIX_YAW_IMAX)
+local last_telem_send = 0.0
+
+function update()
+   -- get the current tilt
+   local tilt_scaled = SRV_Channels:get_output_scaled(k_motor_tilt)
+   if not tilt_scaled then
+      return
+   end
+
+   -- get the current body frame yaw rate
+   local ygyro_rads = ahrs:get_gyro():z()
+   local roll_trate, pitch_trate, yaw_trate = AC_AttitudeControl:get_rpy_rate()
+   local yaw_pid_out = yaw_pid.update(yaw_trate, ygyro_rads)
+
+   -- tilt from 0 to 1
+   local tilt = tilt_scaled * 0.001
+
+   -- get fixed wing aileron
+   local aileron_scaled = SRV_Channels:get_output_scaled(k_aileron)
+   if not aileron_scaled then
+      gcs:send_text(0,string.format("no aileron function", idx))
+      return
+   end
+
+   -- aileron from -1 to 1
+   local aileron = aileron_scaled / 4500.0
+
+   -- get VTOL yaw outputs, sum of FF, P, I and D
+   local yaw_total = yaw_pid_out
+
+   -- tilt angle in radians, 0 means VTOL mode, 1 means fixed wing fwd
+   local tilt_angle_rad = math.pi * 0.5 * tilt
+
+   -- output surface from -1 to 1
+   -- note minus sign on yaw for the frame change
+   local surface = constrain(math.cos(tilt_angle_rad) * (-yaw_total) + math.sin(tilt_angle_rad) * aileron, -1, 1)
+
+   SRV_Channels:set_angle(k_scripting1, 4500)
+   SRV_Channels:set_output_scaled(k_scripting1, surface * 4500)
```

Comment:
Since were now float scaled you should be able to set angle 1 and drop the 4500 multiplier. I have been wanting to do that across the code but have not been brave enough to take it on yet.